### PR TITLE
[SubtitleFormat] - Update SON

### PR DESCRIPTION
**Pull request by ivandrofly**

### CODE REVIEW

File: libse/SubtitleFormats/Son.cs [hunk 1]
```diff
@@ -59,13 +61,14 @@ public override void LoadSubtitle(Subtitle subtitle, List<string> lines, string
...
             subtitle.Paragraphs.Clear();
             _errorCount = 0;
-            var regexTimeCodes = new Regex(@"^\d\d\d\d[\t]+\d\d:\d\d:\d\d:\d\d\t\d\d:\d\d:\d\d:\d\d\t.+\.(tif|tiff|png|bmp|TIF|TIFF|PNG|BMP)", RegexOptions.Compiled);
+
             int index = 0;
+            Match match = null;
             foreach (string line in lines)
             {
-                if (regexTimeCodes.IsMatch(line))
+                if (line.Length > 32 && (match = RegexTimeCodes.Match(line)).Success)
```
Comment (niksedk) [inline libse/SubtitleFormats/Son.cs]: Looks okay - but could you move the assignment to it's own line?

Comment (ivandrofly) [inline libse/SubtitleFormats/Son.cs]: > Looks okay - but could you move the assignment to it's own line?

Sure. you mean like:

``` C#
    if (line.Length > 32)
    {
        match = RegexTimeCodes.Match(line);
        if (match.Success)
        {...
```

Comment (niksedk) [inline libse/SubtitleFormats/Son.cs]: :)

**Comment by ivandrofly** [comment]
Done :wink:

**Pull request closed without merging**